### PR TITLE
Add more clear warning message instructing users how to enable reusable containers

### PR DESCRIPTION
**Pull request by aguibert**
Follow-on subset of the changes proposed in https://github.com/testcontainers/testcontainers-java/pull/2416

CC @bsideup for review

### CODE REVIEW

File: core/src/main/java/org/testcontainers/containers/GenericContainer.java [hunk 1]
```diff
@@ -379,7 +379,8 @@ private void tryStart(Instant startedAt) {
                     }
                     reusable = true;
                 } else {
-                    logger().info("Reuse was requested but the environment does not support the reuse of containers");
+                    logger().warn("Reuse was requested but the environment does not support the reuse of containers");
+                    logger().warn("To enable reuse of containers, you must set 'testcontainers.reuse.enable=true' in a file located at ~/.testcontainers.properties");
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/containers/GenericContainer.java]: Maybe we should make it a bit Windows friendlier? :)

```suggestion
                    logger().warn("To enable reuse of containers, you must set 'testcontainers.reuse.enable=true' in a file located at $HOME/.testcontainers.properties");
```

Comment (bsideup) [reply core/src/main/java/org/testcontainers/containers/GenericContainer.java]: Also, WDYT about merging both messages?

Comment (aguibert) [reply core/src/main/java/org/testcontainers/containers/GenericContainer.java]: sure, how about I change the wording to:
```java
... in a file located at " + System.getProperty("user.home") + "/.testcontainers.properties
```

Comment (bsideup) [reply core/src/main/java/org/testcontainers/containers/GenericContainer.java]: sure 👍

nit: you can use `located at {}/.testcontainers.properties", System.getProperty("user.home")`

Comment (aguibert) [reply core/src/main/java/org/testcontainers/containers/GenericContainer.java]: ok, pushed and ready for review

**Pull request merged**